lib/terraforming/resource/security_group2.rb

```diff
@@ -223,6 +239,153 @@ def tags_attributes_of(security_group)
+        # note: if a rule has multiple security groups then multiple resources will be generated
+        resources = {}
+        nested_rules[rule_type].each do |rule_id, rule|
+
+          # skip the rule count
+          next if rule_id == '#'
+
+          # initialise the rule attributes
+          common_attributes = {
+            'security_group_id': security_group.group_id,
+            'type': rule_type
+          }
+
+          # handle all the simnple parameters
+          ['from_port', 'to_port', 'protocol', 'self', 'protocol'].each do |param|
+            if rule.key?(param)
+              common_attributes[param] = rule[param]
+            end
+          end
+
+          # handle parameters that may have multiple values
+          ['cidr_blocks', 'prefix_list_ids'].each do |param|
+            if rule[param]['#'] != '0'
+              rule[param].each do |key, value|
+                common_attributes["#{param}.#{key}"] = value
+              end
+            end
+          end
+          
+          rule_count = 0
+          sg_resource_name = "aws_security_group.#{module_name_of(security_group)}"
+          # handle security groups
+          if rule['security_groups'].delete('#') != '0'
+            rule['security_groups'].each do |key, source_sg_id|
+              sg_rule_id = "sgrule-#{key}"
+              extra_attributes = {
+                'source_security_group_id': source_sg_id,
+                'id': sg_rule_id
+              }
+
+              resources[rule_resource_name(security_group, rule_type, rule_count)] = {
+                'type': 'aws_security_group_rule',
+                'depends_on': [
+                  sg_resource_name
+                ],
+                'primary': {
+                  'id': sg_rule_id,
+                  'attributes': common_attributes.merge(extra_attributes)
+                }
+              }
+              rule_count += 1
+            end
+          else
+            sg_rule_id = "sgrule-#{rule_id}"
+            extra_attributes = {
+              'id': sg_rule_id
+            }
+
+            resources[rule_resource_name(security_group, rule_type, rule_count)] = {
+              'type': 'aws_security_group_rule',
+              'depends_on': [
+                sg_resource_name
+              ],
+              'primary': {
+                'id': sg_rule_id,
+                'attributes': common_attributes.merge(extra_attributes)
+              }
+            }
+          end
+        end
+        resources
+      end
+
+      def rule_resource_name(security_group, rule_type, rule_count)
+        case rule_type
+          when 'egress'
+            "aws_security_group_rule.#{egress_name_of(security_group, rule_count)}"
+          when 'ingress'
+            "aws_security_group_rule.#{ingress_name_of(security_group, rule_count)}"
+        end
+      end
+
+      def dotted_path_to_hash(hash)
+
+        new_hash = {}
+
+          hash.each do |key, value|
```

Comment:
you don't have to change indent here.

Reply:
In my defence, I copied/pasted that code ;)